### DISCUSSION TITLE
Release candidate FreshRSS 1.23.0

### DISCUSSION THREAD

**Discussion by Alkarex** [FreshRSS/FreshRSS#5963] [Show and tell]
Hello fellow FreshRSS users and contributors,
It looks like we might be able to have a new FreshRSS release 🎁 under the Christmas tree 🎄 for those with that tradition.

So please go ahead and upgrade your [git /edge branch](https://github.com/FreshRSS/FreshRSS), or [Docker :edge image](https://hub.docker.com/repository/docker/freshrss/freshrss), or [download an archive from edge](https://github.com/FreshRSS/FreshRSS/archive/edge.zip), and report any issue or comment, and also when it *just works* 😃

🙋 Please reach out before the 24th for any outstanding bug you might have found or be aware of, in particular if it is a regression compared to 1.22.1.

A few highlights ✨:

* New *Important feeds* group in the main view, with corresponding new priority level for feeds [#5782](https://github.com/FreshRSS/FreshRSS/pull/5782)
    * Entries from important feeds are not marked as read during *scroll*, during *focus*, nor during *Mark all as read*
* Add filter actions (auto mark as read) at category level and at global levels [#5942](https://github.com/FreshRSS/FreshRSS/pull/5942)
* Increase SQL fields length to maximum possible [#5788](https://github.com/FreshRSS/FreshRSS/pull/5788), [#5570](https://github.com/FreshRSS/FreshRSS/pull/5570)
* Many bug fixes

Breaking changes 💥:

* Require PHP 7.4+
* Soft require Apache 2.4+ (but repair minimal compatibility with Apache 2.2)
* Use GitHub Actions to build Docker images, offering architectures `amd64`, `arm32v7`, `arm64v8` with automatic detection [#5808](https://github.com/FreshRSS/FreshRSS/pull/5808)
    * So no `-arm` suffix anymore
* Upgraded extensions require FreshRSS 1.23.0+ [Extensions#181](https://github.com/FreshRSS/Extensions/pull/181)

20 contributors for this release, including several newcomers such as @b-reich, @EdJoPaTo, @FireFingers21, @jaden, @jtracey, @KasimirCash, @mark-monteiro, @foux, @passbe, @joestump ... Thank you!

---

Detailed [tracked changes](https://github.com/FreshRSS/FreshRSS/compare/1.22.1...edge) (20 contributors, 78 commits, 420 changed files with 4,416 additions and 3,520 deletions).

Full [changelog](https://github.com/FreshRSS/FreshRSS/blob/edge/CHANGELOG.md):

* Features
	* New *Important feeds* group in the main view, with corresponding new priority level for feeds [#5782](https://github.com/FreshRSS/FreshRSS/pull/5782)
		* Entries from important feeds are not marked as read during *scroll*, during *focus*, nor during *Mark all as read*
	* Add filter actions (auto mark as read) at category level and at global levels [#5942](https://github.com/FreshRSS/FreshRSS/pull/5942)
	* Improve reliability of *Max number of articles to keep unread* [#5905](https://github.com/FreshRSS/FreshRSS/pull/5905)
	* New option to mark entries as read when focused from keyboard shortcut [5812](https://github.com/FreshRSS/FreshRSS/pull/5812)
	* New display option to hide *My labels* in article footers [#5884](https://github.com/FreshRSS/FreshRSS/pull/5884)
	* Add support for more thumbnail types in feeds enclosures [#5806](https://github.com/FreshRSS/FreshRSS/pull/5806)
	* Support for favicons with non-absolute paths [#5839](https://github.com/FreshRSS/FreshRSS/pull/5839)
	* Increase SQL (`VARCHAR`) text fields length to maximum possible [#5788](https://github.com/FreshRSS/FreshRSS/pull/5788)
	* Increase SQL date fields to 64-bit to be ready for year 2038+ [#5570](https://github.com/FreshRSS/FreshRSS/pull/5570)
* Compatibility
	* Require PHP 7.4+, and implement *typed properties* [#5720](https://github.com/FreshRSS/FreshRSS/pull/5720)
		* Enforce PHP [strict typing](https://php.net/language.types.declarations#language.types.declarations.strict) [#5830](https://github.com/FreshRSS/FreshRSS/pull/5830)
	* Require Apache 2.4+ (but repair minimal compatibility with Apache 2.2) [#5791](https://github.com/FreshRSS/FreshRSS/pull/5791), [#5804](https://github.com/FreshRSS/FreshRSS/pull/5804)
* Bug fixing
	* Fix regression in Docker `CRON_MIN` if any environment variable contains a single quote [#5795](https://github.com/FreshRSS/FreshRSS/pull/5795)
	* Improve filtering of cron environment variables [#5898](https://github.com/FreshRSS/FreshRSS/pull/5898)
	* Fix the `TRUSTED_PROXY` environment variable used in combination with *trusted sources* [#5853](https://github.com/FreshRSS/FreshRSS/pull/5853)
	* Fix regression in marking as read if an identical title already exists [#5937](https://github.com/FreshRSS/FreshRSS/pull/5937)
	* Fix JavaScript regression in label dropdown [#5785](https://github.com/FreshRSS/FreshRSS/pull/5785)
	* Fix regression when renaming a label [#5842](https://github.com/FreshRSS/FreshRSS/pull/5842)
	* Fix API for adding feed with a title [#5868](https://github.com/FreshRSS/FreshRSS/pull/5868)
	* Fix regression in UI of update page [#5802](https://github.com/FreshRSS/FreshRSS/pull/5802)
	* Fix XPath encoding [#5912](https://github.com/FreshRSS/FreshRSS/pull/5912)
* Deployment
	* Use GitHub Actions to build Docker images, offering architectures `amd64`, `arm32v7`, `arm64v8` with automatic detection [#5808](https://github.com/FreshRSS/FreshRSS/pull/5808)
	* Docker alternative image updated to Alpine 3.19 with PHP 8.2.13 and Apache 2.4.58 [#5383](https://github.com/FreshRSS/FreshRSS/pull/5383)
* Extensions
	* Upgrade extensions code to PHP 7.4+ [#5901](https://github.com/FreshRSS/FreshRSS/pull/5901)
	* Breaking change: upgraded extensions require FreshRSS 1.23.0+ [Extensions#181](https://github.com/FreshRSS/Extensions/pull/181)
	* Pass FreshRSS version to JavaScript client side for extensions [#5902](https://github.com/FreshRSS/FreshRSS/pull/5902)
	* Add GitHub Actions and PHPStan for automatic testing of the Extensions repository [Extensions#185](https://github.com/FreshRSS/Extensions/pull/185)
* API
	* Improve handling of new lines in enclosure descriptions (e.g., YouTube video descriptions) [#5859](https://github.com/FreshRSS/FreshRSS/pull/5859)
* Security
	* Avoid printing exceptions in favicons [#5867](https://github.com/FreshRSS/FreshRSS/pull/5867)
	* Remove unneeded execution permissions on some files [#5831](https://github.com/FreshRSS/FreshRSS/pull/5831)
* UI
	* Ensure that enough articles are loaded on window resize [#5815](https://github.com/FreshRSS/FreshRSS/pull/5815)
	* Improve *Nord* theme [#5885](https://github.com/FreshRSS/FreshRSS/pull/5885)
	* Do not show message *Add some feeds* [#5827](https://github.com/FreshRSS/FreshRSS/pull/5827)
	* Various UI and style improvements [#5886](https://github.com/FreshRSS/FreshRSS/pull/5886)
* i18n
	* Fix font priority for languages using Han characters [#5930](https://github.com/FreshRSS/FreshRSS/pull/5930)
	* Improve Dutch [#5796](https://github.com/FreshRSS/FreshRSS/pull/5796)
	* Improve Hungarian [#5918](https://github.com/FreshRSS/FreshRSS/pull/5918)
* Misc.
	* Increase PHPStan from Level 7 to [level 8](https://phpstan.org/user-guide/rule-levels) [#5946](https://github.com/FreshRSS/FreshRSS/pull/5946)
	* Compatibility PHP 8.2+ for running automated tests [#5826](https://github.com/FreshRSS/FreshRSS/pull/5826)
	* Use PHP [`declare(strict_types=1);`](https://php.net/language.types.declarations#language.types.declarations.strict) [#5830](https://github.com/FreshRSS/FreshRSS/pull/5830)
	* Better stack trace for SQL errors [#5916](https://github.com/FreshRSS/FreshRSS/pull/5916)
	* Code improvements [#5511](https://github.com/FreshRSS/FreshRSS/pull/5511), [#5945](https://github.com/FreshRSS/FreshRSS/pull/5945)
	* Update dev dependencies [#5787](https://github.com/FreshRSS/FreshRSS/pull/5787)

**Comment by Alkarex** [comment]
https://github.com/FreshRSS/FreshRSS/releases/tag/1.23.0

**Reply by beerisgood** [reply]
Can you also update the web updater?

**Reply by Alkarex** [reply]
Yes, we typically do it a few days later. Ping @marienfressinaud

**Reply by beerisgood** [reply]
I understand the reason but it’s annoying for all who doesn’t want or can update manually.

**Reply by Alkarex** [reply]
Docker, git, and Web update using git, are all instantly available. It is only the Web update using ZIP, which comes a bit later

**Reply by beerisgood** [reply]
Yes and the internal updater use the zip version.

**Reply by marienfressinaud** [reply]
@beerisgood You should be able to update now!

Just a friendly reminder that we are releasing this version in our spare time and during holidays. No offence, but I just want to avoid putting too much pressure on our shoulders. FreshRSS is being used more and more, and small individual expectations can become a lot of pressure for us. I understand that you would like to get new versions of FreshRSS faster, but it's never more than a few days (unless there are too many problems). You can also switch to the edge version if you're fine with a less stable version of FRSS (a lot of people use it and it's never broken for too long ;))
